### ISSUE TITLE
linking local files without ?fileID does not work

### ISSUE THREAD

**Issue by mstyp** [nextcloud/text#2774]
**Describe the bug**
When I use any other markdown editer to link to a file, even with same relative file structure, in nextcloud opening the link does not work. 

**To Reproduce**
1. highlight text
2. link to local file in the same directory
3. highlight text
4. click "edit link", remove `?fileId=123`, save
5. click link again, a new tab opens but its empty. 
6. go to markor. click link. works fine. 

**Expected behavior**
click link, new tab opens to correct file. 

**Client details:**
 - OS: debian 11
 - Browser: firefox esr
 - Nextcloud version: 23.0.4 - 23.0.4.1
 - Device: desktop

<details>
<summary>Server details</summary>

**Text app version:** (3.4.1

**Operating system:**

debian 11

**Web server:**
Apache/2.4.53 (Debian) (fpm-fcgi)
**Database:**
mysql 10.5.15
**PHP version:**
8
**Nextcloud version:** (see Nextcloud admin page)
23.0.4 - 23.0.4.1
</details>

<details>
<summary>Logs</summary>

#### Nextcloud log (data/nextcloud.log)

to diffuclt to redact

#### Browser log

dont know how to find

</details>

**Comment by vinicius73** [comment]
Related to: 
- #2378
- #1900

**Comment by marcelklehr** [comment]
Confirmed.

**Comment by marningmol** [comment]
I'm happy to see this marked as high priority. 

I'd like to add a user story of just how vital this is.

I've been championing a switch to Nextcloud at our studio. I thought I had tested markdown capabilities by checking that a document rendered properly, including relative image embeds. Since this worked perfectly, it did not occur to me that normal links could somehow not work.

Right now we're trying to decide whether to give up our migration or go through all our links to insert file ids and ruin the workflow of those that use external editors.

I believe this feature is greatly needed, and hope the team can find time to address it quickly.

**Comment by mejo-** [comment]
Dear @marningmol and @mstyp did you consider to try out the [Nextcloud Collectives app](https://apps.nextcloud.com/apps/collectives)? There, links to pages without `fileId` parameter are supposed to work.

**Comment by mstyp** [comment]
No, the collectives app does not properly open links without without `fileId`. Have you tried it yourself? Select some text in collectives, click the "insert link" button, select "link to local file or foulder", pick the file you want to link to (say, test.md in the same directory). Click it and see that it works

But then go to edit the link (select text, then look up to "update link") and then remove the `?fileId=12345`. When you try to click the link then, it thinks you are trying to go to https://test.md lol. 

Even if it did work, I'm not going to stick all my documents into collectives. Its just not the right tool for that.

**Comment by joshtrichards** [comment]
Related: #2136

**Comment by mejo-** [comment]
This should be fixed nowadays.
